vowpalwabbit/io/include/vw/io/logger.h

```diff
@@ -37,8 +38,8 @@ enum class log_level
 
 log_level get_log_level(const std::string& level);
 
-using logger_output_func_t = void (*)(void*, VW::io::log_level, const std::string&);
-using logger_legacy_output_func_t = void (*)(void*, const std::string&);
+using logger_output_func_t = std::function<void(void*, VW::io::log_level, const std::string&)>;
+using logger_legacy_output_func_t = std::function<void(void*, const std::string&)>;
```

Comment:
Same comment for leaving these as is. These also are  a part of the interface and shouldn't be changed without proper deprecation.